binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/aot/KafkaStreamsBinderRuntimeHints.java

```diff
@@ -0,0 +1,91 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kafka.streams.aot;
+
+import java.util.stream.Stream;
+
+import org.apache.kafka.streams.errors.DefaultProductionExceptionHandler;
+import org.apache.kafka.streams.kstream.GlobalKTable;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.Status;
+
+import org.springframework.aop.framework.AopProxyUtils;
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.ProxyHints;
+import org.springframework.aot.hint.ReflectionHints;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeReference;
+import org.springframework.cloud.stream.binder.kafka.streams.GlobalKTableBoundElementFactory;
+import org.springframework.cloud.stream.binder.kafka.streams.KStreamBoundElementFactory;
+import org.springframework.cloud.stream.binder.kafka.streams.KTableBoundElementFactory;
+import org.springframework.lang.Nullable;
+
+
+/**
+ * {@link RuntimeHintsRegistrar} for the Kafka Streams binder in Spring Cloud Stream.
+ *
+ * @author Soby Chacko
+ * @since 4.1.0
+ */
+public class KafkaStreamsBinderRuntimeHints implements RuntimeHintsRegistrar {
+
+	@Override
+	public void registerHints(RuntimeHints hints, @Nullable ClassLoader classLoader) {
+		// The following Kafka Streams specific (3rd party) hints will be removed
+		// once these are added to https://github.com/oracle/graalvm-reachability-metadata
+		registerKafkaStreamsReflectionHints(hints);
+		registerKafkaStreamsJniHints(hints);
+		hints.resources().registerPattern("*/kafka-streams-version.properties");
+
+		// Binder specific hints
+		ProxyHints proxyHints = hints.proxies();
+		registerSpringJdkProxy(proxyHints, KStreamBoundElementFactory.KStreamWrapper.class, KStream.class);
+		registerSpringJdkProxy(proxyHints, KTableBoundElementFactory.KTableWrapper.class, KTable.class);
+		registerSpringJdkProxy(proxyHints, GlobalKTableBoundElementFactory.GlobalKTableWrapper.class, GlobalKTable.class);
+	}
+
+	private static void registerKafkaStreamsJniHints(RuntimeHints hints) {
+		hints.jni().registerType(RocksDBException.class, MemberCategory.DECLARED_FIELDS, MemberCategory.INVOKE_DECLARED_METHODS);
+		hints.jni().registerType(Status.class, MemberCategory.DECLARED_FIELDS, MemberCategory.INVOKE_DECLARED_METHODS);
+
+		hints.resources().registerPattern("librocksdbjni-osx-arm64.jnilib");
+
+	}
+
+	private static void registerKafkaStreamsReflectionHints(RuntimeHints hints) {
+		ReflectionHints reflectionHints = hints.reflection();
+
+		Stream.of(
+				DefaultProductionExceptionHandler.class,
```

Comment:
I suggest to use FQCN for third-parties, so when we got them contributed upstream, it will be much easier to clean up this class.

Reply:
Is this going to be addressed?
Or you think my request is redundant?